python/Auth0/main.py

```diff
@@ -2,70 +2,86 @@
 from scramjet.streams import Stream
 import requests
 import json
+from shiftarray import ShiftArray
 
-class ShiftArray:
-	def __init__(self):
-		self.array = []
+provides = {"provides": "pipe", "contentType": "text/plain"}
 
-	def append(self, value):
-		if len(self.array) == 5:
-			self.array.pop(-1)
-		self.array.append(value)
 
-	def contains(self, value):
-		return value in self.array
+class Auth0:
+    def __init__(self, verified_url, users_url, api_url, request_data, stream):
+        self.verified_url = verified_url
+        self.users_url = users_url
+        self.api_url = api_url
+        self.request_data = request_data
+        self.token_header = {
+            "content-type": "application/json",
+        }
+        self.stream = stream
 
-	def get_values(self):
-		return self.array
+    def lookahead(self, iterable):
+        it = iter(iterable)
+        last = next(it)
+        for val in it:
+            yield last, True
+            last = val
+        yield last, False
 
+    async def get_auth(self):
+        last_verified = ""
+        last_user = ""
+        buffer_verified = ShiftArray()
+        buffer_users = ShiftArray()
+        response = requests.post(
+            self.api_url, headers=self.token_header, data=self.request_data
+        )
+        token = json.loads(response.text)["access_token"]
+        while True:
+            headers = {"authorization": f"Bearer {token}"}
+            verified = requests.get(self.verified_url, headers=headers).json()
+            users = requests.get(self.users_url, headers=headers).json()
+            if "error" in str(verified):
+                response = requests.post(
+                    self.api_url, headers=self.token_header, data=self.request_data
+                )
+                token = json.loads(response.text)["access_token"]
+                continue
```

Comment:
Please add a `asyncio.sleep(WAIT_TIME_ERROR)` here - hold back on error (also define a constant near the top of the file).